src/handlers/oidc.rs

```diff
@@ -115,21 +118,48 @@ broker_handler!(Auth, |app, req| {
     }
 
     if app.providers.contains_key(&email_addr.domain.to_lowercase()) {
-
         // OIDC authentication. Redirect to the identity provider.
         let auth_url = oidc_bridge::request(app, email_addr, client_id, nonce, &parsed_redirect_uri)?;
         Ok(Response::with((status::SeeOther, modifiers::Header(Location(auth_url.to_string())))))
 
     } else {
+        let langtag_de: LanguageTag = "de".parse().unwrap();
+        let mut langtag_en: LanguageTag = Default::default();
+        langtag_en.language = Some("en".to_owned());
+        let langtags = [langtag_de, langtag_en];
+        let catalog_files = ["lang/german.mo", "lang/english.mo"];
+
+        let mut langtag_en_: LanguageTag = Default::default();
+        langtag_en_.language = Some("en".to_owned());
+        let default_accept_language_header = AcceptLanguage(vec![qitem(langtag_en_),]);
+
+        let accept_language_header = req.headers.get::<AcceptLanguage>().unwrap_or(&default_accept_language_header);
+
+        let mut lang_file = "";
+        'outer: for accept_language in accept_language_header.iter() {
+            for (index, langtag) in langtags.iter().enumerate() {
+                if langtag.matches(&accept_language.item) {
+                    lang_file = catalog_files[index];
+                    break 'outer;
+                }
+            }
+        }
+        
+        let f = File::open(lang_file).expect("could not open the catalog");
+        let catalog = Catalog::parse(f).expect("could not parse the catalog");
```

Comment:
I'll repeat the comments from our private discussion on IRC here:

 - For consistency, use `to_string` instead of `to_owned`. (Converting from `&str` to `String`.) Not sure what the Rust preferred option is, but that's what we've used everywhere.

 - LanguageTag implements the `Clone` trait, which'd be a better option than two `langtag_en` variables.

 - I'd opt for keeping the language file names the same as the language tag, which could then simply be mapped to `lang/{}.mo`. We'd still probably want to hardcode the list of files in the code, but it'd be one less 'term' to deal with.

 - We should probably parse catalogs once on init, rather than on every request.

 - Feels like creating a default header value is a roundabout way of just initialising `lang_file` to the default language? When the for-loop falls through, it'd simply use that.

Here's a gist of how I'd write it: https://gist.github.com/stephank/a6c4647d16956335d2f48a62662362f8